### ISSUE TITLE
S3 HeadBucket 301 Error

### ISSUE THREAD

**Issue by daeho-ro** [aws/aws-sdk-java-v2#5085]
### Describe the bug

When I run the head-bucket command from CLI V2 with the bucket located in `us-east-1`,
```
> aws s3api head-bucket --bucket <bucket> --region us-east-1     
{
    "BucketRegion": "us-east-1",
    "AccessPointAlias": false
}
> aws s3api head-bucket --bucket <bucket> --region ap-northeast-2
{
    "BucketRegion": "us-east-1",
    "AccessPointAlias": false
}
```
both region endpoint returns the correct result.

Now, do it the same thing in java sdk v2,
```
        try (S3Client s3Client = s3ClientBuilder.region(Region.US_EAST_1).build()) {
            region1 = s3Client.headBucket(h -> h.bucket(bucket)).bucketRegion();
            System.out.println(region1);
            // returns us-east-1
        }
        try (S3Client s3Client = s3ClientBuilder.region(Region.AP_NORTHEAST_2).build()) {
            region2 = s3Client.headBucket(h -> h.bucket(bucket)).bucketRegion();
            System.out.println(region2);
            // throws 301 exception
        }
```
the second block throws an exception and so cannot get the correct region.

How do I get the bucket region info without it? `GetBucketLocation` is an legacy alternatives but when I read the [documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html) of it, I noticed that they recommend to move an api to `HeadBucket`!

> We recommend that you use [HeadBucket](https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html) to return the Region that a bucket resides in. For backward compatibility, Amazon S3 continues to support GetBucketLocation.

As far as I know with the error, there is no way to replace the `GetBucketLocation` by `HeadBucket`. Is that correct?

### Expected Behavior

Wherever I call the `HeadBucket`, it should return the bucket region info without exception.

### Current Behavior

When I call the `HeadBucket` through the different region endpoint from the bucket region, it throws an exception with the error code 301.

### Reproduction Steps

Place a bucket anywhere, and call the `HeadBucket`.

```
        try (S3Client s3Client = s3ClientBuilder.region(<where the bucket is not located>).build()) {
            s3Client.headBucket(h -> h.bucket(bucket)).bucketRegion();
            // throws 301 exception
        }
```

### AWS Java SDK version used

2.25.27

### JDK version used

21

### Operating System and version

M3 14.3.1

**Comment by bhoradc** [comment]
Hi @daeho-ro, 

Thank you for submitting the issue. In this case where you have to use cross-region on the S3 client builder, you should be enabling the cross-region access for it. You may refer to [Cross-Region access for Amazon S3](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/s3-cross-region.html) from the developer guide.

By default, the cross-region feature is disabled. To enable it for an S3Client, set `crossRegionAccessEnabled` to true on the S3Client Builder:

```
S3Client s3Client = S3Client.builder()
         .region(Region.AP_NORTHEAST_2)
         .crossRegionAccessEnabled(true)
         .build()
```

Feature available on SDK versions 2.20.111 or higher. Please let me know if you have further questions. 

Regards, 
Chaitanya

**Comment by daeho-ro** [comment]
Never thought the option lol. It is working good. Thanks!!
